### PR TITLE
new source, actions, and updated tests

### PR DESCRIPTION
**Pull request by littledice**
This adds methods to get a onetime token for uploading model run parameters.

### CODE REVIEW

File: app/src/main/contrib/sources/ParametersTokenSource.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    constructor() {
+        super();
+        this._fetchOneTimeParametersToken = () => this.doFetch(state => {
+            let queryString = "";
+            if (state.redirectPath && state.redirectPath.length > 0) {
+                queryString = "?redirectUrl=" + encodeURI(settings.montaguUrl() + state.redirectPath);
+            }
+            return `/modelling-groups/${state.currentModellingGroup.id}/model-run-parameters/`
+                + `${state.currentTouchstone.id}/get_onetime_link/${queryString}`;
+        }, {
```
Comment (MartinEden) [inline app/src/main/contrib/sources/ParametersTokenSource.ts]: Something weird with the indentation here

File: app/src/main/contrib/sources/ParametersTokenSource.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+    _fetchOneTimeParametersToken: () => SourceModel<string>;
+
+    constructor() {
+        super();
+        this._fetchOneTimeParametersToken = () => this.doFetch(state => {
+            let queryString = "";
+            if (state.redirectPath && state.redirectPath.length > 0) {
+                queryString = "?redirectUrl=" + encodeURI(settings.montaguUrl() + state.redirectPath);
+            }
```
Comment (MartinEden) [inline app/src/main/contrib/sources/ParametersTokenSource.ts]: The redirect path code is duplicated from elsewhere - could you refactor to this to be more DRY?

File: app/src/main/contrib/sources/ParametersTokenSource.ts (hunk 1 above)
Comment (littledice) [reply app/src/main/contrib/sources/ParametersTokenSource.ts]: how would you format this? this is how my IDE wants to do it...

Comment (MartinEden) [reply app/src/main/contrib/sources/ParametersTokenSource.ts]: Oh, I realise now that I was misreading the code. But that makes me think even more that this isn't the clearest layout. How about [like this](https://github.com/vimc/montagu-webapps/commit/0d7833d7ed4b3d92df48b2a43c221ae47b6313e5)

**Pull request merged**